tests/subsys/net/lib/nrf_cloud/fota_common/src/main.c

```diff
@@ -0,0 +1,823 @@
+
+	zassert_equal(-ENODEV, ret,
+		      "return should be -ENODEV when job type is NRF_CLOUD_FOTA_TYPE__INVALID");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process succeeds when a pending
+ * APP FOTA validation PASSES.
+ */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_pending_fota_job_process_app_pass)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_MCUBOOT_IMG_MANAGER);
+	struct nrf_cloud_settings_fota_job job;
+	bool reboot_required;
+
+	job.validate = NRF_CLOUD_FOTA_VALIDATE_PENDING;
+	job.type = NRF_CLOUD_FOTA_APPLICATION;
+
+	boot_is_img_confirmed_fake.custom_fake = fake_boot_is_img_confirmed__false;
+	boot_write_img_confirmed_fake.custom_fake = fake_boot_write_img_confirmed__succeeds;
+
+	int ret = nrf_cloud_pending_fota_job_process(&job, &reboot_required);
+
+	zassert_equal(0, ret, "return should be 0 when app FOTA succeeds");
+	zassert_equal(NRF_CLOUD_FOTA_VALIDATE_PASS, job.validate,
+		      "validate status should be PASS on success");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process succeeds when a pending
+ * APP FOTA validation FAILS.
+ */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_pending_fota_job_process_app_fail)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_MCUBOOT_IMG_MANAGER);
+	struct nrf_cloud_settings_fota_job job;
+	bool reboot_required = false;
+
+	job.validate = NRF_CLOUD_FOTA_VALIDATE_PENDING;
+	job.type = NRF_CLOUD_FOTA_APPLICATION;
+
+	boot_is_img_confirmed_fake.custom_fake = fake_boot_is_img_confirmed__false;
+	boot_write_img_confirmed_fake.custom_fake = fake_boot_write_img_confirmed__fails;
+
+	int ret = nrf_cloud_pending_fota_job_process(&job, &reboot_required);
+
+	zassert_equal(0, ret, "return should be 0 when app FOTA is processed as failure");
+	zassert_equal(true, reboot_required,
+		      "reboot_required should be true when app FOTA is processed as failure");
+	zassert_equal(NRF_CLOUD_FOTA_VALIDATE_FAIL, job.validate,
+		      "validate status should be FAIL when app FOTA is processed as failure");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process succeeds when a pending
+ * APP FOTA validation is UNKNOWN.
+ */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_pending_fota_job_process_app_unknown)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_MCUBOOT_IMG_MANAGER);
+	struct nrf_cloud_settings_fota_job job;
+	bool reboot_required;
+
+	job.validate = NRF_CLOUD_FOTA_VALIDATE_PENDING;
+	job.type = NRF_CLOUD_FOTA_APPLICATION;
+
+	boot_is_img_confirmed_fake.custom_fake = fake_boot_is_img_confirmed__true;
+
+	int ret = nrf_cloud_pending_fota_job_process(&job, &reboot_required);
+
+	zassert_equal(0, ret, "return should be 0 when app FOTA is already confirmed");
+	zassert_equal(NRF_CLOUD_FOTA_VALIDATE_UNKNOWN, job.validate,
+		      "validate status should be UNKNOWN on when write is already confirmed");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process succeeds when a pending
+ * APP FOTA validation is UNKNOWN because CONFIG_MCUBOOT_IMG_MANAGER is not enabled.
+ */
```

Comment:
```suggestion
/* Verify that nrf_cloud_pending_fota_job_process succeeds when and sets the job validation status to UNKNOWN if
 * provided a pending APP FOTA job and CONFIG_MCUBOOT_IMG_MANAGER is not enabled.
 */
 ```
 
 Also, can you explain (to me at least, not sure yet if it needs to be in the test comments) why UNKNOWN is expected in this case?

Reply:
https://github.com/nrfconnect/sdk-nrf/blob/2006366ca2bc8437bbebbcb7072413abe064dd3f/subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_common.c#L170